test/integration/concrete/lockup/withdraw/withdraw.tree

```diff
@@ -52,46 +52,49 @@ withdraw.t.sol
                       │                ├── it should call the sender hook
                       │                ├── it should emit a {MetadataUpdate} event
                       │                └── it should emit a {WithdrawFromLockupStream} event
-                      └── when the caller is an approved third party
+                      ├── when the caller is an approved third party
+                      │  ├── it should make the withdrawal
+                      │  └── it should update the withdrawn amount
+                      ├── when the caller is an unknown address
```

Comment:
Since we agreed to define **authorized** as someone who is authorized to withdraw to any address, the following nested branches cannot fall under the branch "when the caller is authorized":

- when the caller is an unknown address
- when the caller is the sender

But rather, they should fall under "when the caller is unauthorized".

I am highly of the opinion that we introduce new modifiers: `givenWithdrawalAddressIsRecipient` and `givenWithdrawalAddressNotRecipient` and restructure the tree and the tests. LMK if you agree, I can push those changes and we can review if it looks better or not.

Reply:
Yes. Let's introduce these new modifiers.